### PR TITLE
Forge 1.18.1 Support

### PR DESCRIPTION
**Pull request by LiskoSlayer63**
Support for Forge based servers

### CODE REVIEW

**Review by TBlueF — CHANGES_REQUESTED** [review]
Build is not working ^^ (https://github.com/BlueMap-Minecraft/BlueMap/runs/5018286595)
These Java-Version problems are the reason why i did not add Forge 1.18 support yet ..
I got it working on the `feature/v2` branch tho.
The plan is to wait for that to release and release forge 1.18 support with that :)

**Comment by LiskoSlayer63** [comment]
@TBlueF It needs Java 17 to compile, Forge demands it for MC 1.18 and newer. That caused me a lot of head scratching and that's the reason why also `com.github.johnrengelman.shadow` was bumped to `7.1.2` from `6.1.0`. 

The whole project compiles succesfully with Java 17 after updating the Gradle Shadow plugin.

**Comment by TBlueF** [comment]
Alright seems like this is working :)
I'll not merge this into master tho, because it will create a lot of merge-conflicts with my `feature/v2` branch once i want to merge that. Since `feature/v2` already contains a working forge 1.18.1 build :)
I'll pull it into a temporary extra branch for now and  upload the new forge-build to curseforge :)
I hope that's fine ^^

**Pull request merged**
